File: Intro/code.js
// Time Complexity Examples in JavaScript

// Constant Time Complexity = O(1)
let x = 5;
console.log(x);

if (x === 5) {
  console.log("hello");
} else {
  console.log("world");
}

// Linear Time Complexity = O(n)
x = 5;
for (let i = 0; i < x; i++) {
  console.log(i);
}

// Quadratic Time Complexity = O(n^2)
x = 5;
for (let i = 0; i < x; i++) {
  for (let j = 0; j < x; j++) {
    console.log("hello");
  }
}

for (let k = 0; k < x; k++) {
  console.log("world");
}

// Cubic Time Complexity = O(n^3) (or O(n^2 * m))
const n = 5;
const m = 10;
for (let i = 0; i < n; i++) {
  for (let j = 0; j < m; j++) {
    for (let k = 0; k < n; k++) {
      console.log("hello");
    }
  }
}

// Logarithmic Time Complexity = O(log n)
let N = 10;
while (N >= 1) {
  console.log("hello");
  N = Math.floor(N / 2);
}

File: Intro/code.py
# Time Complexity Examples

# Constant Time Complexity = O(1) program example
x = 5
print(x)

if (x == 5):
    print("hello")
else:
    print("world")

# Linear Time Complexity = O(n) program example    
x = 5 
for i in range(0, x):
    print(i)

# Quadratic Time Complexity = O(n^2) program example
x = 5
for i in range(0, x):
    for j in range(0, x):
        print("hello")

for k in range(0, x):
    print("world")

# Cubic Time Complexity = O(n^3) program example
n = 5
m = 10
for i in range(0, n):             # this loop run n times
    for j in range(0, m):         # this loop run m times
        for k in range(0, n):     # this loop run n times
            print("hello")        # this loop run n * m * n times = O(n^2 * m) = O(n^3) 

# Logarithmic Time Complexity = O(log n) program example     
n =10 
while(n >= 1): 
    print("hello")
    n = n // 2      
